usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -115,73 +72,171 @@ def delete_docs_by_unique_key(client: Elasticsearch, key: str, value_list: list,
         config (dict): collection of key-value pairs that encapsulates runtime arguments for this ES management task
         index (str): Optional name, alias, or pattern of index this query will target. Looks up via config if not
-                     provided
+            provided
+        lookup_chunk_size (int): the batch-size of terms value-array to be looked-up. Needs to be less
+            than 65536 (max values for any terms query), and less than config["max_query_size"]
 
-    Returns: None
+    Returns: list of values for the ES_AWARDS_UNIQUE_KEY_FIELD fields in the looked-up documents.
     """
     if index is None:
         index = f"{config['query_alias_prefix']}-*"
-    col_to_items_dict = defaultdict(list)
-    for line in id_list:
-        col_to_items_dict[line["col"]].append(line["key"])
-    awards = []
-    for column, values in col_to_items_dict.items():
-        values_generator = chunks(values, 1000)
-        for v in values_generator:
-            body = filter_query(column, v)
-            response = client.search(index=index, body=json.dumps(body), size=config["max_query_size"])
-            if response["hits"]["total"]["value"] != 0:
-                awards += [x["_source"][ES_AWARDS_UNIQUE_KEY_FIELD] for x in response["hits"]["hits"]]
-    return awards
-
-
-def delete_awards(client: Elasticsearch, config: dict) -> None:
+
+    if not _is_allowed_key_field_type(client, lookup_key, index):
+        msg = (
+            f'Cannot perform lookups in index "{index}" with key field "{lookup_key}" because its type is not one of '
+            f"the allowed field types, or the field was not found in that index."
+        )
+        logger.error(format_log(msg=msg, action="Delete"))
+        raise RuntimeError(msg)
+
+    if lookup_chunk_size > 65536:
+        # 65,536 is max number of terms that can be added to an ES terms filter query
+        msg = (
+            f"{lookup_chunk_size} is greater than 65,536, which is the max number of terms that can be added to an ES "
+            f"terms filter query"
+        )
+        logger.error(format_log(msg=msg, action="Delete"))
+        raise RuntimeError(msg)
+
+    if lookup_chunk_size > config["max_query_size"]:
+        # Some keys would be left undiscovered if our chunk was cut short by the query only returning a lesser subset
+        msg = (
+            f"{lookup_chunk_size} is greater {config['max_query_size']}, which is the max number of query "
+            f"results returnable from this index. Use a smaller chunk or increase max_result_window for this index."
+        )
+        logger.error(format_log(msg=msg, action="Delete"))
+        raise RuntimeError(msg)
+
+    award_key_list = []
+    values_generator = chunks(value_list, lookup_chunk_size)
+    for chunk_of_values in values_generator:
+        q = Search(using=client, index=index).filter("terms", **{lookup_key: chunk_of_values})  # type: Search
```

Comment:
is `# type: Search` helpful information here?

Reply:
It's a form of type hint convention introduced by [PEP 484](https://www.python.org/dev/peps/pep-0484/#type-comments). Supporting IDEs like PyCharm will then know what type is returned by the last function in a chained call like that, and then provide api-autocomplete as well as static code analysis thereafter.